app/_src/mesh/guides/create-a-mesh-using-terraform-and-konnect.md

```diff
@@ -0,0 +1,602 @@
+
+Create a new k3d cluster:
+```bash
+k3d cluster create tfmink
+```
+
+Write `tfmink` cluster configuration:
+```bash
+export KUBECONFIG=$(k3d kubeconfig write tfmink)
+```
+
+Update the `main.tf` file to include the time, kubernetes and helm provider:
+
+```hcl
+terraform {
+  required_providers {
+    konnect = {
+      source = "kong/konnect"
+    }
+    konnect-beta = {
+      source  = "kong/konnect-beta"
+      version = "{{ TF_BETA_VERSION }}"
+    }
+    time = {
+      source  = "hashicorp/time"
+      version = "{{ TF_TIME_VERSION }}"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "{{ TF_KUBERNETES_VERSION }}"
+    }
+    helm = {
+      source  = "hashicorp/helm"
+      version = "{{ TF_HELM_VERSION }}"
+    }
+  }
+}
+```
+
+Add a variable pointing to the Kubeconfig file: 
+
+```hcl
+variable "k8s_cluster_config_path" {
+  type        = string
+  description = "The location where this cluster's kubeconfig will be saved to."
+  default     = "~/.config/k3d/kubeconfig-tfmink.yaml"
+}
+```
+
+You can override that variable by setting the environment variable `TF_VAR_k8s_cluster_config_path`.
+
+We need to create a system account and a token to authenticate the zone:
+
+```hcl
+resource "konnect_system_account" "zone_system_account" {
+  depends_on = [konnect_mesh_control_plane.my_meshcontrolplane]
+  name            = "mesh_${konnect_mesh_control_plane.my_meshcontrolplane.id}_${var.zone_name}"
+  description     = "Terraform generated system account for authentication zone ${var.zone_name} in ${konnect_mesh_control_plane.my_meshcontrolplane.id} control plane."
+  konnect_managed = false
+}
+
+resource "konnect_system_account_role" "zone_system_account_role" {
+  depends_on = [konnect_system_account.zone_system_account]
+  account_id       = konnect_system_account.zone_system_account.id
+  entity_id        = konnect_mesh_control_plane.my_meshcontrolplane.id
+  entity_region    = var.region
+  entity_type_name = "Mesh Control Planes"
+  role_name        = "Admin" # should be "Connector"
+}
+
+resource "time_offset" "one_year_from_now" {
+  offset_years = 1
+}
+
+resource "konnect_system_account_access_token" "zone_system_account_token" {
+  depends_on = [konnect_system_account.zone_system_account, time_rotating.rotate, time_offset.one_year_from_now]
+  account_id = konnect_system_account.zone_system_account.id
+  expires_at = time_offset.one_year_from_now.rfc3339
+  name       = konnect_system_account.zone_system_account.name
+}
+```
+
+We need to create a namespace and a secret to hold the Zone token:
+
+```hcl
+resource "kubernetes_namespace" "kong_mesh_system" {
+  metadata {
+    name = "kong-mesh-system"
+    labels = {
+      "kuma.io/system-namespace" = "true"
+    }
+  }
+}
+
+resource "kubernetes_secret" "mesh_cp_token" {
+  metadata {
+    name = "cp-token"
+    namespace = "kong-mesh-system"
```

Comment:
namespace probably should be `kubernetes_secret.kong_mesh_system.metadata.name`

Reply:
yes, but:

```
    namespace = kubernetes_namespace.kong_mesh_system.metadata.name
```

Reply:
actually:

```suggestion
    namespace = kubernetes_namespace.kong_mesh_system.metadata.0.name
```